src/components/map/Map.jsx

```diff
@@ -54,4 +52,29 @@ const Map = compose(
   </GoogleMap>
 ));
 
+class Map extends Component {
```

Comment:
Is this new component necessary?
I personally like standard components like this way more than all the recompose stuff, but having only the toggling in this format, and all the rest in the recompose style, feels inconsistent.

Reply:
Yes. There were some conflicts in state change between the stateHandler methods and the lifecycle methods.
Also, I needed a reference to the map component, which is not possible with a recompose HOC.

Reply:
OK, if there were state conflicts this might be easier.

As for a ref, that actually works with recompose too - ultimately you can just treat it as any callback prop, that's just happens to be called with the instance. (As you can see, onCenterChanged was already using a ref; exposing it to the parent is just a matter of also invoking a callback.)